### PR TITLE
add test to make sure managedFields can be reset

### PR DESCRIPTION
**Pull request by kwiesmueller**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR goes hand in hand with a PR to kubernetes/website in order to assure and document `managedFields` can be reset/deleted using `MergePatch`:
https://github.com/kubernetes/website/pull/14300

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig apimachinery
/wg apply
/cc @apelisse @jennybuckley

### CODE REVIEW

**Comment by neolit123** [comment]
the new test LGTM.

File: test/integration/apiserver/apply/apply_test.go [hunk 1]
```diff
@@ -830,3 +830,59 @@ func TestApplyConvertsManagedFieldsVersion(t *testing.T) {
...
+		Get()
+	if err != nil {
+		t.Fatalf("Failed to create object using Apply patch: %v", err)
+	}
+
+	_, err = client.CoreV1().RESTClient().Patch(types.MergePatchType).
+		Namespace("default").
+		Resource("configmaps").
+		Name("test-cm").
+		Body([]byte(`{"metadata":{"managedFields": [{}]}}`)).Do().Get()
```
Comment (apelisse) [inline test/integration/apiserver/apply/apply_test.go]: Do you need the empty dictionary in the lisit, or would this also work: `{"metadata":{"managedFields": []}}`
And what about: `{"metadata":{"managedFields": null}}`

**Review by apelisse — COMMENTED** [review]
One question, looks good otherwise

File: test/integration/apiserver/apply/apply_test.go (hunk 1 above)
Comment (kwiesmueller) [reply test/integration/apiserver/apply/apply_test.go]: I haven't tried null, good point.
Using an empty array does not work as we have a check in the FieldManager to ensure we don't reset on an empty array.
It's in the update function, I can post a link tomorrow

**Pull request merged**